airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresCatalogHelper.java

```diff
@@ -61,6 +58,42 @@ public static AirbyteStream overrideSyncModes(final AirbyteStream stream) {
     return stream.withSupportedSyncModes(Lists.newArrayList(SyncMode.FULL_REFRESH, SyncMode.INCREMENTAL));
   }
 
+  public static List<String> getViewsInSchema(final JdbcDatabase database, final String schema) throws SQLException {
+    final String query = "SELECT viewname FROM pg_catalog.pg_views WHERE schemaname = ?";
```

Comment:
I think this should work, but the existing solution is to check for filenode of the stream. Maybe we should consider consolidate them? 

See `supportResumableFullRefresh` in `PostgresSource.java`

Reply:
@xiaohansong and I had a discussion and we decide to keep the current implementation. We will also file another ticket to track a potential improvement we could do on this file node function that issues a db query to each stream.